### ISSUE TITLE
Resource 'aws_elasticache_cluster.redis' does not have attribute 'cache_nodes.*.address' for variable 'aws_elasticache_cluster.redis.cache_nodes.*.address'

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#214]
_This issue was originally opened by @iroller as hashicorp/terraform#7957. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version

`0.7.0`
### Affected Resource(s)
- aws_elasticache_cluster
- module outputs
### Terraform Configuration Files

``` hcl

## Module
resource "aws_elasticache_cluster" "redis" {
  engine               = "redis"
...
}

// Addresses
output "addresses" {
  value = "${join(",", aws_elasticache_cluster.redis.cache_nodes.*.address)}"
}


## Using module
module "redis" {
...
}

output "redis_endpoint_addresses" {
  value = "${module.redis.addresses}"
}
```
### Debug Output

```
[WARN] Output interpolation "addresses" failed: Resource 'aws_elasticache_cluster.redis' does not have attribute 'cache_nodes.*.address' for variable 'aws_elasticache_cluster.redis.cache_nodes.*.address'
```
### Expected Behavior

I expected a string outout, e.g. "10.10.0.1,10.10.0.2"
### Actual Behavior

No output at all.
### References
- https://github.com/hashicorp/terraform/issues/4703 - similar, but not sure if it's the same issue

**Comment by sstarcher** [comment]
I assumed I could work around this using ```output nodes {
  value = "${var.replicas == "1" ?
      aws_elasticache_cluster.cache.cache_nodes.0.address :
      join(",",aws_elasticache_cluster.cache.cache_nodes.*.address)
  }"
}
```

but no luck anytime I reference `aws_elasticache_cluster.cache.cache_nodes.*.address` it drops the output

**Comment by fstuck37** [comment]
Hi All,
Was wondering if anyone had an update on this?
Thanks,
Fred

**Comment by inthedeepend** [comment]
I also get no output with this, any updates/fixes on the horizon? Workarounds would also be appreciated.

**Comment by pberswe** [comment]
I have the same issue as described

**Comment by adybuxton** [comment]
I am also seeing a similar issue and i've raised a seperate issue here https://github.com/hashicorp/terraform/issues/19964

**Comment by admssa** [comment]
The funny thing - next construction works for other resources but not for this one:
```
output "redis_endpoint" {
  value = "${var.registry_enabled ?  aws_elasticache_cluster.docker-registry-cache.cache_nodes.0.address : "nope" }"```

it returns
```Resource 'aws_elasticache_cluster.docker-registry-cache' does not have attribute 'cache_nodes.0.address' for variable 'aws_elasticache_cluster.docker-registry-cache.cache_nodes.0.address'```

**Comment by bflad** [comment]
Hi folks 👋 This issue is resolved in Terraform 0.12, which fully supports indexed splat (`*`) notation with list attribute values.

Given this configuration:

```hcl
terraform {
  required_providers {
    aws = "2.20.0"
  }
  required_version = "0.12.5"
}

provider "aws" {
  region = "us-east-2"
}

resource "aws_elasticache_cluster" "test" {
  cluster_id      = "bflad-testing"
  engine          = "memcached"
  node_type       = "cache.t2.micro"
  num_cache_nodes = 3
}

output "test1" {
  value = aws_elasticache_cluster.test.cache_nodes[*].address
}

output "test2" {
  value = element(aws_elasticache_cluster.test.cache_nodes[*].address, 1)
}

output "test3" {
  value = join(",", formatlist("%s:%s", aws_elasticache_cluster.test.cache_nodes[*].address, aws_elasticache_cluster.test.cache_nodes[*].port))
}
```

We can get the various expected outputs that were problematic in Terraform 0.11 and earlier:

```
Outputs:

test1 = [
  "bflad-testing.asxu6y.0001.use2.cache.amazonaws.com",
  "bflad-testing.asxu6y.0002.use2.cache.amazonaws.com",
  "bflad-testing.asxu6y.0003.use2.cache.amazonaws.com",
]
test2 = bflad-testing.asxu6y.0002.use2.cache.amazonaws.com
test3 = bflad-testing.asxu6y.0001.use2.cache.amazonaws.com:11211,bflad-testing.asxu6y.0002.use2.cache.amazonaws.com:11211,bflad-testing.asxu6y.0003.use2.cache.amazonaws.com:11211
```

Enjoy! 🚀
